### DISCUSSION TITLE
enable proxmox

### DISCUSSION THREAD

**Discussion by ale82x** [librenms/docker#443] [Q&A]
Hello,
i have a librenms on docker.
on docs i read that i edit file config.php
https://docs.librenms.org/Extensions/Proxmox/

but i don't have file config.php on docker compose.

i add some rows on .env file?
but i can't find syntax...

someone can help me?
thank you

**Comment by LoveSkylark** [comment]
config.php is out dated and is should be faces out, all the config now reside in the DB and creating a config.php file means hard overwriting the current settings (but this can sometime be useful where you want to make sure no one can mess some settings up) . 

You can use "lnms config:set" instead inside opt/librenms

So:
$config['enable_proxmox'] = 1;

Translates to:
lnms config:set enable_proxmox 1

You can also check the config state before you do any change by doing "lnms config:get enable_proxmox"

With this in mind should be able to construct a docker compose "command:" that can execute the  configuration during the startup process instead of having to volume map in a "config.php" file

**Comment by rjadams82** [comment]
except in docker this results in 

```shell
root@server:~# docker exec ef5e9edd8421 lnms config:set enable_proxmox 1
This is not a valid setting. Please check your input
```

**Comment by zndrr** [comment]
Experiencing the same issue here. Syntax doesn't appear to be available on the Docker instance lnms context.

```
user@lnmshost:~$ lnms config:get enable_proxmox

user@lnmshost:~$ lnms config:set enable_proxmox 1
This is not a valid setting. Please check your input
```
Note that I tried a lot of combinations of the same.

However, after dumping an arbitrarily named  `proxmox.php` file in the `./config` dir, following [THIS](https://github.com/librenms/docker?tab=readme-ov-file#live-config) instruction, with the following content:
```
<?php
$config['enable_proxmox'] = 1;
```
It seemed to have registered the setting

```
user@lnmshost:~$ lnms config:get enable_proxmox
1
```
However, the set option was still unavailable.

**Comment by murrant** [comment]
That is an upstream bug.  In order for it to validate correctly with config:set, it must be defined in https://github.com/librenms/librenms/blob/master/misc/config_definitions.json (which is still incomplete).

You can add --ignore-checks to set "any" settting even ones with typoes ;)

**Reply by rjadams82** [reply]
Silly me, reading the docs. If only I'd come here first. 

> You can add --ignore-checks to set "any" settting even ones with typoes ;)

@murrant you are a true gentleman. 

```shell
root@server:~# docker exec ef5e9edd8421 lnms config:set enable_proxmox 1
This is not a valid setting. Please check your input

root@server:~# docker exec ef5e9edd8421 lnms config:set enable_proxmox 1 --ignore-checks 

root@server:~# docker exec ef5e9edd8421  lnms config:get enable_proxmox
1

root@server:~# 
``` 

Fixed for me.

**Reply by murrant** [reply]
https://github.com/librenms/librenms/pull/17032
